pkg/policy/policy_controller.go

```diff
@@ -566,6 +570,69 @@ func getTrigger(rd kyverno.ResourceDescription) []*kyverno.ResourceSpec {
 	return specs
 }
 
+func generateTriggers(client client.Interface, rule kyverno.Rule, log logr.Logger) []*kyverno.ResourceSpec {
+	var specs []*kyverno.ResourceSpec
+	list := &unstructured.UnstructuredList{}
+
+	for _, kind := range rule.MatchResources.Kinds {
+		mlist, err := client.ListResource("", kind, "", rule.MatchResources.Selector)
+		if err != nil {
+			log.Error(err, "failed to list matched resource")
+		}
+		list.Items = append(list.Items, mlist.Items...)
+	}
+
+	for _, any := range rule.MatchResources.Any {
+		for _, kind := range any.Kinds {
+			anyList, err := client.ListResource("", kind, "", any.Selector)
+			if err != nil {
+				log.Error(err, "failed to list any matched resource")
+			}
+			list.Items = append(list.Items, anyList.Items...)
+		}
+	}
+
+	var kindlist []string
+	for _, all := range rule.MatchResources.All {
+		kindlist = append(kindlist, all.Kinds...)
+	}
+
+	if isMatchResourcesAllValid(kindlist) {
```

Comment:
Can we return the matching kind from `isMatchResourcesAllValid` and use the returned kind to list resources? Iterating through `match.all.resources.kinds` introduces unnecessary calls and could result in throttling issues.

Reply:
Passed the unique list kinds if its a valid all kinds